integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioSessionMigrIstioGateway.java

```diff
@@ -0,0 +1,460 @@
+
+    return istioIngressPort;
+  }
+
+  private Map<String, String> processHttpRequest(String curlUrlPath,
+                                                 String headerOption) {
+    String[] httpAttrArray = {"sessioncreatetime", "sessionid", "primary", "secondary", "count"};
+    Map<String, String> httpAttrInfo = new HashMap<String, String>();
+
+    // build curl command
+    String curlCmd = buildCurlCommand(curlUrlPath, headerOption);
+    logger.info("==== Command to set HTTP request and get HTTP response {0} ", curlCmd);
+
+    // set HTTP request and get HTTP response
+    ExecResult execResult = assertDoesNotThrow(
+        () -> execCommand(domainNamespace, adminServerPodName,
+        null, true, "/bin/sh", "-c", curlCmd));
+    if (execResult.exitValue() == 0) {
+      logger.info("\n ===== HTTP response is \n " + execResult.stdout());
+      assertAll("Check that primary server name is not null or empty",
+          () -> assertNotNull(execResult.stdout(), "Primary server name shouldn’t be null"),
+          () -> assertFalse(execResult.stdout().isEmpty(), "Primary server name shouldn’t be  empty")
+      );
+
+      for (String httpAttrKey : httpAttrArray) {
+        String httpAttrValue = getHttpResponseAttribute(execResult.stdout(), httpAttrKey);
+        httpAttrInfo.put(httpAttrKey, httpAttrValue);
+      }
+    } else {
+      fail("==== Failed to process HTTP request " + execResult.stderr());
+    }
+
+    return httpAttrInfo;
+  }
+
+  private String buildCurlCommand(String curlUrlPath, String headerOption) {
+    final String httpHeaderFile = "/u01/domains/header";
+    final String clusterAddress = domainUid + "-cluster-" + clusterName;
+    logger.info("Build a curl command with pod name {0}, curl URL path {1} and HTTP header option {2}",
+        clusterAddress, curlUrlPath, headerOption);
+
+    int waittime = 5;
+    return new StringBuilder()
+        .append("curl --silent --show-error")
+        .append(" --connect-timeout ").append(waittime).append(" --max-time ").append(waittime)
+        .append(" http://")
+        .append(K8S_NODEPORT_HOST)
+        .append(":")
+        .append(istioIngressPort)
+        .append("/")
+        .append(curlUrlPath)
+        .append(headerOption)
+        .append(httpHeaderFile).toString();
+  }
+
+  private String getHttpResponseAttribute(String httpResponseString, String attribute) {
+    // retrieve the search pattern that matches the given HTTP data attribute
+    String attrPatn = httpAttrMap.get(attribute);
+    assertNotNull(attrPatn,"HTTP Attribute key shouldn’t be null");
+
+    // search the value of given HTTP data attribute
+    Pattern pattern = Pattern.compile(attrPatn);
+    Matcher matcher = pattern.matcher(httpResponseString);
+    String httpAttribute = null;
+
+    if (matcher.find()) {
+      httpAttribute = matcher.group(2);
+    }
+
+    return httpAttribute;
+  }
+
+  private void shutdownServerUsingServerStartPolicy(String msName) {
```

Comment:
It can be moved to a utility class.

Reply:
changed

Reply:
Should it be a utility in CommonTestUtils instead of SessionMigrationUtil class, since this has nothinh to do with SessionMigration usecase

Reply:
We already have a mathod   public static boolean shutdownManagedServerUsingServerStartPolicy in TestAction.java. shutdownServerUsingServerStartPolicy in SessionMigrationUtil is misnamed. Let me change the name

Reply:
There are two calls in 
1. call TestAction.shutdownManagedServerUsingServerStartPolicy
2. varify servers are shutted down.

So I change the name of shutdownServerUsingServerStartPolicy to shutdownServerAndVerify. Since we have to shutdown the server to test session migration, I feel it's proper to have it in SessionMigrationUtil.java